Error running Quarkus: java.lang.ExceptionInInitializerError   > @POST with Record

### Describe the bug

Quarkus 3.30.2 on Java 21

I have a Rest endpoint (tried with an without @BeanParam, same result):

```
 @POST
    @Produces("image/png")
    @Consumes(MediaType.APPLICATION_JSON)
    public Response generate(@BeanParam QRCodeConfig cfg) {
```

And the

```
public record QRCodeConfig(
        String text,
        Integer size,

```
And I get a weird error starting my application:

```
6:53:34,858 ERROR [io.qua.run.Quarkus] Error running Quarkus: java.lang.ExceptionInInitializerError
    at java.base/jdk.internal.misc.Unsafe.allocateInstance(Native Method)
    at java.base/java.lang.invoke.DirectMethodHandle.allocateInstance(DirectMethodHandle.java:501)
    at io.quarkus.runtime.Quarkus.run(Quarkus.java:79)
    at io.quarkus.runtime.Quarkus.run(Quarkus.java:51)
    at io.quarkus.runtime.Quarkus.run(Quarkus.java:144)
    at io.quarkus.runner.GeneratedMain.main(Unknown Source)
    at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
    at java.base/java.lang.reflect.Method.invoke(Method.java:580)
    at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:135)
    at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.RuntimeException: Failed to start quarkus
    at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
    ... 10 more
Caused by: java.lang.VerifyError: Bad type on operand stack
Exception Details:
  Location:
    com/example/QRCodeConfig.__quarkus_rest_inject(Lorg/jboss/resteasy/reactive/server/injection/ResteasyReactiveInjectionContext;)Lcom/example/QRCodeConfig; @31: invokespecial
  Reason:
    Type uninitialized 27 (current frame, stack[1]) is not assignable to 'java/lang/Integer'
  Current Frame:
    bci: @31
    flags: { }
    locals: { 'org/jboss/resteasy/reactive/server/injection/ResteasyReactiveInjectionContext', null, null, null, null, null, null, null, null, null, null }
    stack: { uninitialized 27, uninitialized 27 }
  Bytecode:
    0000000: 014c 014d 014e 013a 0401 3a05 013a 0601
    0000010: 3a07 013a 0801 3a09 013a 0abb 0001 59b7
    0000020: 0063 b0

    at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
    at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3578)
    at java.base/java.lang.Class.getMethodsRecursive(Class.java:3719)
    at java.base/java.lang.Class.getMethod0(Class.java:3705)
    at java.base/java.lang.Class.getMethod(Class.java:2393)
    at org.jboss.resteasy.reactive.server.core.parameters.RecordBeanParamExtractor.<init>(RecordBeanParamExtractor.java:16)
    at org.jboss.resteasy.reactive.server.core.startup.RuntimeResourceDeployment.parameterExtractor(RuntimeResourceDeployment.java:708)
    at org.jboss.resteasy.reactive.server.core.startup.RuntimeResourceDeployment.buildResourceMethod(RuntimeResourceDeployment.java:346)
    at org.jboss.resteasy.reactive.server.core.startup.RuntimeDeploymentManager.deploy(RuntimeDeploymentManager.java:137)
    at io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveRecorder.createDeployment(ResteasyReactiveRecorder.java:163)
    at io.quarkus.runner.recorded.ResteasyReactiveProcessor$setupDeployment944680882.deploy_0(Unknown Source)
    at io.quarkus.runner.recorded.ResteasyReactiveProcessor$setupDeployment944680882.deploy(Unknown Source)
```



### Expected behavior

It works :)

### Actual behavior

Exception during startup.

### How to Reproduce?

I have a reproducer here:
https://github.com/myfear/ejq_substack_articles/tree/main/qr-designer

Just `quarkus dev` and see the exception. 
mvn compile works.

### Output of `uname -a` or `ver`

Darwin Kernel Version 25.1.0

### Output of `java -version`

OpenJDK Runtime Environment Temurin-21.0.8+9 (build 21.0.8+9-LTS)

### Quarkus version or git rev

3.30.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.11 (3e54c93a704957b63ee3494413a2b544fd3d825b)

### Additional information

_No response_

--- a/extensions/resteasy-reactive/rest/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/beanparam/EmptyBeanParamRecordTest.java
+++ b/extensions/resteasy-reactive/rest/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/beanparam/EmptyBeanParamRecordTest.java
@@ -1,0 +1,45 @@
+package io.quarkus.resteasy.reactive.server.test.beanparam;
+
+import jakarta.ws.rs.BeanParam;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class EmptyBeanParamRecordTest {
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest()
+            .setArchiveProducer(() -> {
+                return ShrinkWrap.create(JavaArchive.class)
+                        .addClasses(EmptyBeanParam.class, Resource.class);
+            }).assertException(x -> {
+                x.printStackTrace();
+                Assertions.assertEquals(
+                        "Body parameters (or non-annotated fields) are not allowed for records. Make sure to annotate your record components with @Rest* or @*Param or that they can be injected as context objects.",
+                        x.getMessage());
+            });
+
+    @Test
+    void empty() {
+        Assertions.fail();
+    }
+
+    public record EmptyBeanParam(String something, Integer other) {
+    }
+
+    @Path("/")
+    public static class Resource {
+
+        @Path("/record")
+        @GET
+        public String beanParamRecord(@BeanParam EmptyBeanParam param) {
+            return "OK";
+        }
+    }
+}

--- a/extensions/resteasy-reactive/rest/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/beanparam/ReallyEmptyBeanParamRecordTest.java
+++ b/extensions/resteasy-reactive/rest/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/beanparam/ReallyEmptyBeanParamRecordTest.java
@@ -1,0 +1,46 @@
+package io.quarkus.resteasy.reactive.server.test.beanparam;
+
+import jakarta.ws.rs.BeanParam;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class ReallyEmptyBeanParamRecordTest {
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest()
+            .setArchiveProducer(() -> {
+                return ShrinkWrap.create(JavaArchive.class)
+                        .addClasses(EmptyBeanParam.class, Resource.class);
+            }).assertException(x -> {
+                x.printStackTrace();
+                Assertions.assertEquals(
+                        "No annotations found on fields at 'io.quarkus.resteasy.reactive.server.test.beanparam.ReallyEmptyBeanParamRecordTest$EmptyBeanParam'. Annotations like `@QueryParam` should be used in fields, not in methods.",
+                        x.getMessage());
+            });
+
+    @Test
+    void empty() {
+        Assertions.fail();
+    }
+
+    // This one does not even have body params
+    public record EmptyBeanParam() {
+    }
+
+    @Path("/")
+    public static class Resource {
+
+        @Path("/record")
+        @GET
+        public String beanParamRecord(@BeanParam EmptyBeanParam param) {
+            return "OK";
+        }
+    }
+}

--- a/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/ServerEndpointIndexer.java
+++ b/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/ServerEndpointIndexer.java
@@ -395,6 +395,10 @@ public class ServerEndpointIndexer
                     annotations, field.type(), "%s", new Object[] { field }, applyFieldRules, hasRuntimeConverters,
                     // We don't support annotation-less path params in injectable beans: only annotations
                     Collections.emptySet(), field.name(), EMPTY_STRING_ARRAY, new HashMap<>());
+            if (currentClassInfo.isRecord() && result.getType() == ParameterType.BODY) {
+                throw new DeploymentException(
+                        "Body parameters (or non-annotated fields) are not allowed for records. Make sure to annotate your record components with @Rest* or @*Param or that they can be injected as context objects.");
+            }
             if ((result.getType() != null) && (result.getType() != ParameterType.BEAN)) {
                 //BODY means no annotation, so for fields not injectable
                 fieldExtractors.put(field, result);

